bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManagerImpl.java

```diff
@@ -1116,6 +1144,40 @@ private void setThingStatus(Thing thing, ThingStatusInfo thingStatusInfo) {
         }
     }
 
+    @Override
+    public void setEnabled(ThingUID thingUID, boolean enabled) {
+        Thing thing = getThing(thingUID);
+
+        if (enabled) {
+            // Enable a thing
+            logger.debug("Thing {} will be enabled.", thing.getUID().getAsString());
+            if (storage != null) {
+                storage.remove(thingUID.getAsString());
+            }
+            if (!thing.getStatus().equals(ThingStatus.ONLINE)) {
```

Comment:
What is the purpose of this condition? Do you maybe mean `if (!ThingHandlerHelper.isHandlerInitialized(thing)) {...}`?